### PR TITLE
cpp-sort: add version 1.14.0, remove older versions

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **cpp-sort/***

- add version 1.14.0
- use `add_subdirectory` in test_v1_package
- remove older versions
- use `target_compile_features`
- sort and add quote in conandata.yml
- improve compiler version checks

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by toge** [comment]
@Morwenn 
Thank you for developing cpp-sort and creating recipe.

I just happened to find the cpp-sort release via the explorer function on github.
I'd like to follow the author's idea as much as possible, so if it is better to wait for the project to be updated, I will do so next time. Please let me know.

It is a conan-center-index approved process to delete some older versions.
https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#removing-old-versions

I'm trying to save time for CI by reducing the number of versions.
Even if I remove a version from a recipe, you can continue to get the old version from conan-center, so there should be no inconvenience.

**Comment by Morwenn** [comment]
I don't have a problem with someone more knowledgeable than me handling the Conan packaging. My library isn't really doing anything complicated in its build process, so it's absolutely fine if someone else does it - I'll just need to check that someone hasn't already opened a PR in the future 🙂 

Thanks for the extra information!

**Comment by SpaceIm** [comment]
I would advice to remove old versions after conan migration, so that they benefit to at least one RREV supporting conan v2, unless it's too much work or burns c3i, but here it's quite fast to build in c3i.

File: recipes/cpp-sort/all/conanfile.py [hunk 1]
```diff
@@ -89,8 +99,5 @@ def package(self):
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "cpp-sort"
         self.cpp_info.names["cmake_find_package_multi"] = "cpp-sort"
```
Comment (SpaceIm) [inline recipes/cpp-sort/all/conanfile.py:101]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "cpp-sort")
        self.cpp_info.set_property("cmake_target_name", "cpp-sort::cpp-sort")
```

**Comment by toge** [comment]
@SpaceIm 
I agree with you.
Let me revert this recipe.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**